core/nodejsActionBase/runner.js

```diff
@@ -71,13 +71,20 @@ function NodeActionRunner(message, whisk) {
         var result = this.userScriptMain(args);
 
         if (result !== this.whisk.async()) {
-            // This happens, e.g. if you just have "return;"
-            if (typeof result == 'undefined') {
-                result = {};
-            }
-            closeAndReturn(result);
+            var self = this;
+            Promise.resolve(result) //Non-promises/undefined instantly resolve.
+                .then(function(resolvedResult) {
+                    // This happens, e.g. if you just have "return;"
+                    if (typeof resolvedResult === 'undefined') {
+                        resolvedResult = {};
+                    }
+                    closeAndReturn(resolvedResult);
+                }, function (error) {
+                        self.whisk.error(error);
```

Comment:
yes, there is an action container test in this PR that rejects the Promise and will hit this path.